syntax = "proto2";

message LoginData {
  required string login = 1;
  required string password = 2;
}

message LoginResponse {
	enum Status {
		OK = 0;
		INVALID_CREDIENTIALS = 1;
	}
	
	required string token = 1;
	required Status status = 2;
}

message Token {
  required string token = 1;
}

message Permissions {
  required string mask = 1;
}

message Empty {
}

service AuthService {
   rpc GetToken(LoginData) returns (LoginResponse);
   rpc InvalidateToken(Token) returns (Empty);
   rpc GetPermissions(Token) returns (Permissions);
}
